src/nodedef.cpp

```diff
@@ -586,6 +588,10 @@ void ContentFeatures::deSerialize(std::istream &is)
 
 	try {
 		node_dig_prediction = deSerializeString(is);
+		u8 tmp_leveled_max = readU8(is);
+		if (is.eof()) /* readU8 doesn't throw exceptions so we have to do this */
```

Comment:
What if `readU8` reads the last byte? Shouldn't this be checked before reading the byte?

Reply:
`eof` is only set after readU8 unsuccessfully tried to read the last byte.
At least that's the behaviour I remember, might be wrong.

Reply:
Indeed, you remembered correctly.
Logging before and after `readU8`:
```
before? 0
after? 1
```